server/s3_post_complete_action.cc

```diff
@@ -548,7 +549,7 @@ void S3PostCompleteAction::add_part_object_to_probable_dead_oid_list(
         // bucketing of object)
         S3CommonUtilities::size_based_bucketing_of_objects(
             ext_oid_str, part_entry[0].item_size);
-        if (is_old_object) {
+        if (!is_old_object) {
```

Comment:
if there is old object(overwritre case)
then key is <oldid>-<newoid>

else (no overwrite) then key is newoid